### PR TITLE
ci: platform checks: introduce @disabled annotation, restore ignores

### PR DESCRIPTION
**Pull request by nrainer-materialize**
Restore the possibility to ignore checks.

### CODE REVIEW

File: misc/python/materialize/checks/all_checks/__init__.py [hunk 1]
```diff
@@ -10,11 +10,21 @@
 import glob
 from os.path import basename, dirname, isfile, join
 
+# file names with extension
+IGNORED_CHECK_FILE_NAMES = [
```
Comment (def-) [inline misc/python/materialize/checks/all_checks/__init__.py:14]: As I wrote on Slack, I think it's better to derive the classes from `CheckDisabled`. Then it's explicit at the location of the check that it's disabled, not hidden in yet another location.

**Review by def- — CHANGES_REQUESTED** [review]

**Comment by nrainer-materialize** [comment]
No longer ignoring `statement_logging`, which is fixed with https://github.com/MaterializeInc/materialize/pull/22624.

File: misc/python/materialize/checks/checks.py [hunk 2]
```diff
@@ -20,6 +21,7 @@ class Check:
     def __init__(self, base_version: MzVersion, rng: Random | None) -> None:
         self.base_version = base_version
         self.rng = rng
+        self.disabled = False
```
Comment (def-) [inline misc/python/materialize/checks/checks.py:24]: Now that I think about it, maybe `self.enabled = True` is better than double negation.

**Review by def- — APPROVED** [review]

File: misc/python/materialize/checks/checks.py (hunk 2 above)
Comment (nrainer-materialize) [reply misc/python/materialize/checks/checks.py:24]: Sure, happy to adjust that

**Pull request merged**